### PR TITLE
SLE-80 Replace Builder by a ResourceChangeListener

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintChangeListener.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.runtime.CoreException;
+import org.sonarlint.eclipse.core.internal.jobs.AnalyzeProjectJob;
+import org.sonarlint.eclipse.core.internal.jobs.AnalyzeProjectRequest;
+import org.sonarlint.eclipse.core.internal.resources.SonarLintProject;
+
+public class SonarLintChangeListener implements IResourceChangeListener {
+
+  @Override
+  public void resourceChanged(IResourceChangeEvent event) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintChangeListener.java]: MAJOR The Cyclomatic Complexity of this method "resourceChanged" is 16 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintChangeListener.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+              // Workspace root
+              return true;
+            }
+            final SonarLintProject sonarProject = SonarLintProject.getInstance(delta.getResource().getProject());
+            if (!sonarProject.isAutoEnabled()) {
+              return false;
+            }
+            if (delta.getResource().getType() == IResource.FILE) {
+              IFile file = (IFile) delta.getResource();
+              if (delta.getKind() == IResourceDelta.CHANGED
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintChangeListener.java]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintChangeListener.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import org.sonarlint.eclipse.core.internal.resources.SonarLintProject;
+
+public class SonarLintChangeListener implements IResourceChangeListener {
+
+  @Override
+  public void resourceChanged(IResourceChangeEvent event) {
+    if (event.getType() == IResourceChangeEvent.POST_CHANGE) {
+      final Map<IProject, Collection<IFile>> changedFilesPerProject = new HashMap<>();
+      try {
+        event.getDelta().accept(new IResourceDeltaVisitor() {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintChangeListener.java]: MAJOR Reduce this anonymous class number of lines from 26 to at most 20, or make it a named class. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1188)
MAJOR Make this anonymous inner class a lambda [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1604)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**